src/framework/mpas_attlist.F

```diff
@@ -253,6 +261,213 @@ subroutine mpas_add_att_text(attList, attName, attValue, ierr)!{{{
+        if (cursor % attType == MPAS_ATT_INTA) then
+          if (present(ierr)) ierr = 0
+          cursor % attValueIntA(:) = attValue(:)
+        end if
+        return
+      end if
+      cursor => cursor % next
+    end do
+
+  end subroutine mpas_modify_att_int1d!}}}
+
+!***********************************************************************
+!
+! routine mpas_modify_att_real0d
+!
+! > \brief  MPAS modify 0D real attribute routine
+! > \author Matthew Dimond
+! > \date   06/27/23
+! > \details
+! > This routine modifies a 0d real attribute in the attribute list,
+! > and returns a 1 in ierr if the attribute is not found, or the attribute
+! > has a type incompatible with attValue.
+!
+!----------------------------------------------------------------------
+  subroutine mpas_modify_att_real0d(attList, attName, attValue, ierr)!{{{
+
+    implicit none
+
+    type (att_list_type), pointer :: attList !< Input/Output: Attribute List
+    character (len=*), intent(in) :: attName !< Input: Att. name to modify
+    real (kind=RKIND), intent(in) :: attValue !< Input: Updated Att. value
+    integer, intent(out), optional :: ierr !< Output: Error flag
+
+    type (att_list_type), pointer :: cursor
+
+    if (present(ierr)) ierr = 1
+
+    ! Traverese list looking for attName
+    cursor => attlist
+    do while (associated(cursor))
+      if (trim(cursor % attName) == trim(attName)) then
+        if (cursor % attType == MPAS_ATT_REAL) then
+          if (present(ierr)) ierr = 0
+          cursor % attValueReal = attValue
+        end if
+        return
+      end if
+      cursor => cursor % next
+    end do
+
+  end subroutine mpas_modify_att_real0d!}}}
+
+!***********************************************************************
+!
+! routine mpas_modify_att_real1d
+!
+! > \brief  MPAS modify 1D real attribute routine
+! > \author Matthew Dimond
+! > \date   06/27/23
+! > \details
+! > This routine modifies a 1d real attribute in the attribute list,
+! > and returns a 1 in ierr if the attribute is not found, or the attribute
+! > has a type incompatible with attValue.
+!
+!----------------------------------------------------------------------
+  subroutine mpas_modify_att_real1d(attList, attName, attValue, ierr)!{{{
+
+    implicit none
+
+    type (att_list_type), pointer :: attList !< Input/Output: Attribute List
+    character (len=*), intent(in) :: attName !< Input: Att. name to modify
+    real (kind=RKIND), dimension(:), intent(in) :: attValue !< Input: Updated Att. value
+    integer, intent(out), optional :: ierr !< Output: Error flag
+
+    type (att_list_type), pointer :: cursor
+
+    if (present(ierr)) ierr = 1
+
+    ! Traverese list looking for attName
+    cursor => attlist
+    do while (associated(cursor))
+      if (trim(cursor % attName) == trim(attName)) then
+        if (cursor % attType == MPAS_ATT_REALA) then
+          if (present(ierr)) ierr = 0
+          cursor % attValueRealA(:) = attValue(:)
```

Comment:
Here and also in the `int1d` routine, it may be good to also verify that the size of the `attValue` array matches the size of the `attValueRealA` array.